store.go

```diff
@@ -1932,13 +1944,50 @@ func (s *store) Mount(id, mountLabel string) (string, error) {
 	if modified, err := rlstore.Modified(); modified || err != nil {
 		rlstore.Load()
 	}
-	if rlstore.Exists(id) {
-		return rlstore.Mount(id, mountLabel)
+	if !rlstore.Exists(id) {
+		return "", ErrLayerUnknown
 	}
-	return "", ErrLayerUnknown
+
+	rootUID, rootGID, err := idtools.GetRootUIDGID(s.uidMap, s.gidMap)
+	if err != nil {
+		return "", err
+	}
+
+	merged := ""
+	if rootUID != 0 {
+		merged = filepath.Join(s.RunRoot(), id, "merged")
+		if err = idtools.MkdirAllAs(merged, 0700, rootUID, rootGID); err != nil {
+			return "", err
+		}
+		// runc doesn't like symlinks in the rootfs, let's make it happy
+		merged, err = filepath.EvalSymlinks(merged)
+		if err != nil {
+			return "", errors.Wrapf(err, "cannot resolve symlink %s", merged)
+		}
+	}
+
+	mount, err := rlstore.Mount(id, mountLabel)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if err != nil {
```

Comment:
This won't read the result of the `unix.Mount()` call below, since the scope of that variable is the `if` statement that defines it.